ratpack-core/src/main/java/ratpack/http/client/HttpClient.java

```diff
@@ -106,8 +112,20 @@ static HttpClient httpClient(ServerConfig serverConfig, Registry registry) {
    * @param maxContentLengthBytes The max content length of a response to support.
    * @return An instance of a HttpClient
    */
-  static HttpClient httpClient(ByteBufAllocator byteBufAllocator, int maxContentLengthBytes) {
-    return new DefaultHttpClient(byteBufAllocator, maxContentLengthBytes);
+  static HttpClient httpClient(ByteBufAllocator byteBufAllocator, int maxContentLengthBytes, ExecController execController) {
```

Comment:
What would a sane default for ExecController be?

We didn't want to make this change, honestly, but we were unable to find a way to to build a client without knowing the ExecController at bind time, since the  Netty connection pools a bootstrap that depends on getting a handle to the event loop group.